### PR TITLE
doc: add a change log/release notes draft file

### PR DESCRIPTION
**Pull request by ru-fu**
Add a file where changes between releases can be tracked.
This file mostly links to change logs (on the wiki and in
sdk-nrfxlib), but will eventually contain content for
sdk-zephyr.

Signed-off-by: Ruth Fuchss <ruth.fuchss@nordicsemi.no>

### CODE REVIEW

**Comment by ru-fu** [comment]
> 
> 
> I'm a bit confused by the approach here.
> 
> I thought we were going to use RST for this instead of the changelog in the future, but it appears that it's an RST file that just points to the changelog.
> 
> Which is it? I don't have a preference, but this seems inconsistent with what I remember us discussing on Monday.

The discussion on Monday was only about the Zephyr part of the change log - and this is what this PR prepares for (content should be added in lines 27 and 31). 
For general NCS changes and changes in nrfxlib, we have not decided any changes, so general NCS changes will still go into the wiki, and library changes will still be tracked in each library's change log.

**Comment by carlescufi** [comment]
> > I'm a bit confused by the approach here.
> > I thought we were going to use RST for this instead of the changelog in the future, but it appears that it's an RST file that just points to the changelog.
> > Which is it? I don't have a preference, but this seems inconsistent with what I remember us discussing on Monday.
> 
> The discussion on Monday was only about the Zephyr part of the change log - and this is what this PR prepares for (content should be added in lines 27 and 31).
> For general NCS changes and changes in nrfxlib, we have not decided any changes, so general NCS changes will still go into the wiki, and library changes will still be tracked in each library's change log.

While confusing for the users, this is indeed what we agreed upon @mbolivar-nordic. At the end of the release cycle we will move everything from the wiki into this `.rst.` file, whereas the zephyr-specific changes will already be there.

File: doc/nrf/releases/release-notes-latest.rst [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+########################
+
+The most relevant changes that are present on the master branch of the |NCS|, as compared to the latest release, are tracked in this file.
+Note that this file is a work in progress and might not cover all relevant changes.
+
+
+Change log
+**********
+
+See the `master branch section in the change log`_ for a list of the most important changes.
```
Comment (mbolivar-nordic) [inline doc/nrf/releases/release-notes-latest.rst:13]: Since libraries are tracked separately, and Zephyr is tracked here, I think it is important to make it clear that this changelog only covers the nrf repository somehow. Perhaps something like this:

```suggestion
See the `master branch section in the change log`_ for a list of the most important changes in the ``nrf`` repository.
```

File: doc/nrf/releases/release-notes-latest.rst [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+===========
+
+See the change log for each library in the :doc:`nrfxlib documentation <nrfxlib:README>` for the most current information.
+
+sdk-zephyr
+==========
+
+The Zephyr fork in |NCS| contains all commits from the upstream Zephyr repository up to and including ``ac117c8d``, plus some |NCS| specific additions.
+
+The following list summarizes the most important changes from upstream Zephyr:
```
Comment (mbolivar-nordic) [inline doc/nrf/releases/release-notes-latest.rst:25]: The distinction between this bullet list and the second one took me a couple of tries to understand, because "changes from upstream zephyr" could be read as either of these:

1. divergence in NCS's fork of zephyr from the upstream zephyr repository
2. changes in NCS's fork of zephyr in the master branch since the most recent release

On first read, I assumed it was 2., but actually it seems to be 1. after reading the below on line 29. Since the list is likely to grow quite long, I think it is worth it to clarify this here, since line 29 will be line 400 at some point, I am guessing, and will not be visible on screen at the same time.

Is there some way to reword this to be more clear?

**Comment by mbolivar-nordic** [comment]
Thanks @ru-fu and @carlescufi for the clarification. I have some opinions given that decision.

File: doc/nrf/releases/release-notes-latest.rst (hunk 1 above)
Comment (ru-fu) [reply doc/nrf/releases/release-notes-latest.rst:13]: It doesn't - the wiki change log also contains entries for bsdlib and the zboss stack, which are in the nrfxlib repo. It's (currently) intended to cover the NCS.
That's why I mentioned it directly under the "Change log" section and did not make a repo subsection for it.

File: doc/nrf/releases/release-notes-latest.rst (hunk 2 above)
Comment (ru-fu) [reply doc/nrf/releases/release-notes-latest.rst]: Updated to say "The following list summarizes the most important **commits** from upstream Zephyr:".

Comment (carlescufi) [reply doc/nrf/releases/release-notes-latest.rst]: > Updated to say "The following list summarizes the most important **commits** from upstream Zephyr:".

That is still a bit confusing. How about "The following list summarizes the most important changes inherited from upstream Zephyr:" ?

Comment (ru-fu) [reply doc/nrf/releases/release-notes-latest.rst]: Either works. Updated.

File: doc/nrf/releases/release-notes-latest.rst (hunk 1 above)
Comment (mbolivar-nordic) [reply doc/nrf/releases/release-notes-latest.rst:13]: OK, how about "most important changes not covered here" or so then?

Comment (ru-fu) [reply doc/nrf/releases/release-notes-latest.rst:13]: If this is how we plan to use the wiki change log?
My understanding was that the wiki change log is a summary covering all of NCS. And that we need a place ( => this file here) where we can list in more detail what the Zephyr upmerges contain. I would expect though that we mention the upmerge in the wiki as well (since this is where people will probably look first), and then the "not covered here" would be misleading.

Comment (mbolivar-nordic) [reply doc/nrf/releases/release-notes-latest.rst:13]: > My understanding was that the wiki change log is a summary covering all of NCS. And that we need a place ( => this file here) where we can list in more detail what the Zephyr upmerges contain.

Then why is there an nrfxlib entry here too?

That's why I was asking about "not covered here". Since the wiki's changelog seems to cover "everything in NCS that isn't nrfxlib or zephyr", not just "everything in NCS that isn't zephyr".
